Mikem 901 readdata

mmastand opened the pull request:
Thin wrapper to read from a table using a sql query. I opted not to allow `dplyr` code here since the user base of that is much smaller and it's pretty easy to do without any helper functions.

Added:
- `db_read`. This function reads from a database using a supplied connection and query. Pulling the data into memory is optional.

Test:
- Docs, examples, tests
- Try to use it on your client env?

michaellevy commented:
@mmastand -- would you see if you can close #490 as part of this through the check connection functionality you mentioned? Would be a substantial value add to error instead of hanging.

mmastand commented:
Conversation on odbc board about writing to non-default schemas: https://github.com/r-dbi/odbc/issues/91

mmastand commented:
It looks like the new rstudio connection pane takes care of that. I haven't been able to reproduce the error on the new version

mmastand commented:
@michaellevy 
- Fixed logic in that one function
- Added codecov.yml so we could ignore those 2 functions.

mmastand commented:
@michaellevy Ready.

Refactored and cleaned old functions
Build and Read are in `db_connections`.
Tests are in `test-db-connections`.
All write functions were removed.
Old db read and write were depreciated.
Docs updated.
RODBC depend removed.

michaellevy commented on R/db_connections.R:
```diff
@@ -1,263 +1,157 @@
+#' my_con <- build_connection_string(server = "localhost",
+#'                                   database = "myDB")
+#' con <- DBI::dbConnect(odbc::odbc(), .connection_string = cs)
+#' d <- db_read(con, "SELECT * FROM my_table)
```
Missing closing quote here and below

michaellevy commented on R/db_connections.R:
```diff
@@ -1,263 +1,157 @@
-#' head(confirmDf)
+#' my_con <- build_connection_string(server = "localhost",
+#'                                   database = "myDB")
+#' con <- DBI::dbConnect(odbc::odbc(), .connection_string = cs)
```
Shouldn't this be `.connection_string = my_con`?

michaellevy commented on R/db_connections.R:
```diff
@@ -1,263 +1,157 @@
-#' @references \url{http://healthcareai-r.readthedocs.io}
-#' @seealso \code{\link{healthcareai}}
-#' @seealso \code{\link{selectData}}
 #' @examples
```
Let's use a realistic example. Here's what I used to test on HPH if you want to just use this go ahead:

my_con <- build_connection_string(server = "HPHI-EDWDEV")
con <- DBI::dbConnect(odbc::odbc(), .connection_string = my_con)
d <- db_read(con, "SELECT TOP 10 * FROM [Shared].[Cost].[FacilityAccountCost]")

mmastand replied:
Done, switched this out. How self serving! I think it's better this way.

michaellevy commented:
@mmastand I fixed the lints because there were some files that were being ignored by lintr (declared in `.lintr`) that are invovled here, and master will soon contain changes that make travis fail if there are litnts, and I didn't want you to get hit by that. Shouldn't affect you, just be sure to pull at some point (and obviously merge master.)

michaellevy commented:
also, appveyor not being able to build is because of the merge conflict. when you merge master that should be resolved.

mmastand commented:
Thanks. I'll get back to addressing these comments soon.

mmastand commented:
Good review, thanks!

*Why not wrap them together in a `db_connect` function?*
  - If the connection isn't made in the outermost environment, the RStudio connections pane functionality won't work. With just a couple exceptions, packages are staying away from making connections inside functions. I think it's more important to have the connection pane functionality than a `db_connect` function. Too much extra work!
- Merged in master, looks good. Thanks for fixing the lints filling me in on that. 
- `dbplyr` is required for the back end database stuff. I got an error when it wasn't loaded on my VM saying it wasn't there. I checked the Appveyor build with and without the dependency and found that it's needed for Appveyor to pass.
